controllers/kserve_customcacert_controller.go

```diff
@@ -0,0 +1,190 @@
+			return err
+		}
+		log.V(1).Info("Deleted the storage-config Secret to update the custom cert")
+	}
+
+	return nil
+}
+
+// This section is intended for regenerating StorageSecret using new data.
+func (r *KServeCustomCACertReconciler) deleteStorageSecret(ctx context.Context, namespace string) error {
+	foundStorageSecret := &corev1.Secret{}
+
+	err := r.Get(ctx, types.NamespacedName{
+		Name:      constants.DefaultStorageConfig,
+		Namespace: namespace,
+	}, foundStorageSecret)
+
+	if err == nil {
+		err = r.Delete(ctx, foundStorageSecret)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func checkOpenDataHubGlobalCertCAConfigMapName(objectName string) bool {
+	return objectName == odhGlobalCACertConfigMapName
+}
+
+// reconcileOpenDataHubGlobalCertConfigMap filters out all ConfigMaps that are not the OpenDataHub global certificate ConfigMap.
+func reconcileOpenDataHubGlobalCACertConfigMap() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			objectName := e.Object.GetName()
+			return checkOpenDataHubGlobalCertCAConfigMapName(objectName)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			objectName := e.Object.GetName()
+			return checkOpenDataHubGlobalCertCAConfigMapName(objectName)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			objectName := e.Object.GetName()
+			return checkOpenDataHubGlobalCertCAConfigMapName(objectName)
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			objectName := e.ObjectNew.GetName()
+			return checkOpenDataHubGlobalCertCAConfigMapName(objectName)
+		},
+	}
+}
+
+func (r *KServeCustomCACertReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// Initialize logger format
+	log := r.Log.WithValues("ConfigMap", req.Name, "namespace", req.Namespace)
+
+	configmap := &corev1.ConfigMap{}
+	err := r.Get(ctx, req.NamespacedName, configmap)
+	if err != nil && apierrs.IsNotFound(err) {
+		log.Info("Opendatahub global cert ConfigMap not found")
+		configmap.Namespace = req.Namespace
+	} else if err != nil {
+		log.Error(err, "Unable to fetch the ConfigMap")
+		return ctrl.Result{}, err
+	}
+
+	err = r.reconcileConfigMap(configmap, ctx)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *KServeCustomCACertReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Create a builder that only watch OpenDataHub global certificate ConfigMap
+	builder := ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.ConfigMap{}).
+		WithEventFilter(reconcileOpenDataHubGlobalCACertConfigMap())
+	err := builder.Complete(r)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func getDesiredCaCertConfigMapForKServe(configmapName string, namespace string, caCertData map[string]string) *corev1.ConfigMap {
+	desiredConfigMap := &corev1.ConfigMap{
```

Comment:
Could we add some annotation here to represent that this configmap is maintain my `odh-model-controller`

Reply:
I added label `"opendatahub.io/managed": "true"`